addons/product/migrations/9.0.1.2/post-migration.py

```diff
@@ -276,6 +277,79 @@ def update_product_supplierinfo(env):
             AND rc.currency_id != pp.currency_id """)
 
 
+def migrate_purchase_pricelists(env):
+    """ Create product.supplierinfo records for "simple" pricelist items,
+    those are items based on a field (not other pricelist or supplier items)"""
+    env.cr.execute("select id, field from product_price_type")
+    for type_id, field_name in env.cr.fetchall():
+        if not openupgrade.column_exists(env.cr, 'product_template', field_name):
+            continue
+        env.cr.execute(
+            """
+            with partner2pricelist_id as (
+                select
+                    split_part(res_id, ',', 2)::int partner_id,
+                    split_part(value_reference, ',', 2)::int pricelist_id,
+                    company_id
+                from ir_property
+                where
+                fields_id in (
+                    select res_id from ir_model_data
+                    where name='field_res_partner_property_product_pricelist_purchase'
+                ) and
+                res_id is not null and value_reference is not null
+            ),
+            partner2pricelist as (
+                select
+                    partner_id, pp.id pricelist_id, p2p.company_id, currency_id
+                from partner2pricelist_id p2p
+                join product_pricelist pp on p2p.pricelist_id=pp.id and pp.active
+            ),
+            partner2version as (
+                select
+                    partner_id, ppv.id version_id, p2p.company_id,
+                    p2p.currency_id, date_start, date_end
+                from partner2pricelist p2p
+                join product_pricelist_version ppv
+                    on ppv.pricelist_id=p2p.pricelist_id and active
+            ),
+            partner2item as (
+                select
+                    partner_id, ppi.id item_id, p2v.company_id, p2v.currency_id,
+                    p2v.date_start, p2v.date_end, price_discount, price_surcharge,
+                    product_id, product_tmpl_id, min_quantity
+                from partner2version p2v
+                join product_pricelist_item ppi
+                    on ppi.openupgrade_legacy_9_0_price_version_id=p2v.version_id
+                where openupgrade_legacy_9_0_base=%(type_id)s and (
+                    product_id is not null or product_tmpl_id is not null
+                )
+            )
+            insert into product_supplierinfo
+            (
+                name, min_qty, currency_id, date_start, date_end,
+                company_id, product_id, product_tmpl_id, price
+            )
+            select
+            partner_id, min_quantity, currency_id, p2i.date_start, p2i.date_end,
+            p2i.company_id, p2i.product_id, p2i.product_tmpl_id,
+            coalesce(pt1.%(field_name)s, pt2.%(field_name)s) *
+            (1 + price_discount) + price_surcharge
```

Comment:
Instead of `(1 + price_discount)` shouldn't be `(1 - 1 * (price_discount / 100))` ?